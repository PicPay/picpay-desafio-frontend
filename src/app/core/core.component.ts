import { Component } from '@angular/core';

@Component({
  selector: 'core-app',
  template:`
    <app-wrapper></app-wrapper>
  `,
})
export class CoreComponent {
  title = 'Desafio Picpay Front-end';
}
