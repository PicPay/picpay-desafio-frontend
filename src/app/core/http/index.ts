import { BaseService } from "./base.service";
import { UsersService } from "./users/users.service";
import { TransactionService } from "./transaction/transaction.service";

export const httpServices = [BaseService, UsersService, TransactionService];
