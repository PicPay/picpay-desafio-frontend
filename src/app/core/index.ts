export * from './components/index';
export * from './model/index';
export * from './services/index';
