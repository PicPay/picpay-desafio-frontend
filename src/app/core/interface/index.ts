export * from './repositories/itransaction-repository';
export * from './repositories/iusers-repository';

export * from './usecases/itransaction-usecase';
export * from './usecases/iusers-usecase';
