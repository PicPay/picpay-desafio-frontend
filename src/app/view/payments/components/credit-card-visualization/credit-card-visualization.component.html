<app-credit-card [cardInformation]="card"></app-credit-card>
