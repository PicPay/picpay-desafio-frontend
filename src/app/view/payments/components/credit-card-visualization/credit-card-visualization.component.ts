import { Component, Inject } from '@angular/core';
import { MAT_DIALOG_DATA } from '@angular/material/dialog';
import { creditCard } from 'src/app/core/types/credit-card.type';

@Component({
  selector: 'app-credit-card-visualization',
  templateUrl: './credit-card-visualization.component.html',
  styleUrls: ['./credit-card-visualization.component.scss'],
})
export class CreditCardVisualizationComponent {
  constructor(@Inject(MAT_DIALOG_DATA) private data: { card: creditCard }) {}

  get card(): creditCard {
    return this.data.card;
  }
}
