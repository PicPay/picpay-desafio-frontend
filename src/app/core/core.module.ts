import { FormControlValidatorService } from '@core/services/form-control-validator/form-control-validator.service';
import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { ApiService } from '@core/services/api/api.service';
import { ThemeService } from './services/theme/theme.service';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HttpClientModule,
    ApiService,
    ThemeService,
    FormControlValidatorService,
  ],
})
export class CoreModule {}
