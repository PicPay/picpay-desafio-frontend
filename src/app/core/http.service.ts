import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class HttpService {

  constructor(
    private http: HttpClient
  ) { }

  get<T>(path: string): Observable<T> {
    return this.request<T>('get', path);
  }
  post<T>(path: string, options = {}): Observable<T> {
    return this.request<T>('post', path, options);
  }

  private request<T>(requestType: string, path: string, options = {}): Observable<T> {
    const headers = this.getHeaders();
    return this.http.request<T>(requestType, path, { ...options, headers });
  }

  private getHeaders() {
    return new HttpHeaders({
      Authorization: 'Bearer',
    });
  }
}
