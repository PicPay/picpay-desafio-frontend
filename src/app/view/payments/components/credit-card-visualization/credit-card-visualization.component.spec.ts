import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { async, TestBed } from '@angular/core/testing';
import { CreditCardVisualizationComponent } from './credit-card-visualization.component';


describe('CreditCardVisualizationComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      declarations: [CreditCardVisualizationComponent],
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(CreditCardVisualizationComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const fixture = TestBed.createComponent(CreditCardVisualizationComponent);
    const el = fixture.debugElement.nativeElement.outerHTML;
    expect(el).toMatchSnapshot();
  });
});
