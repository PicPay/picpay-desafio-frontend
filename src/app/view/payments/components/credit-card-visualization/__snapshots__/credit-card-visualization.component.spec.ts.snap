// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCardVisualizationComponent should match the snapshot 1`] = `"<div id=\\"root1\\" ng-version=\\"8.2.14\\"><app-credit-card></app-credit-card></div>"`;
